google/cloud/spanner_v1/_pandas_helpers.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import warnings
+
+
+def pd_dataframe(result_set):
+    """This functions converts the query results into pandas dataframe
+
+    :rtype: pandas.Dataframe
+    :returns: Dataframe with the help of a mapping dictionary which maps every spanner datatype to a pandas compatible datatype.
+    """
+    try:
+        from pandas import DataFrame
+    except ImportError as err:
+        raise ImportError(
+            "Pandas module not found. It is needed for converting query result to Dataframe.\n Try running 'pip3 install pandas'"
+        ) from err
+
+    data = list(result_set)
+
+    columns_dict = {}
+    for item in result_set.fields:
+        columns_dict[item.name] = item.type_.code
+
+    # Creating list of columns to be mapped with the data
+    column_list = list(columns_dict.keys())
```

Comment:
As I said before, if we're using a dictionary we can't rely on the keys being ordered until Python 3.7. We should create a list of columns in the loop.

```suggestion
    columns_dict = {}
    column_list = []
    for item in result_set.fields:
        column_list.append(item.name)
        columns_dict[item.name] = item.type_.code
```